fwprovider/resource_options.go

```diff
@@ -633,20 +647,83 @@ func (r *clusterOptionsResource) Update(ctx context.Context, req resource.Update
 	resp.Diagnostics.Append(resp.State.Set(ctx, plan)...)
 }
 
-// Delete deletes a cluster options interface.
-//
-//nolint:lll
-func (r *clusterOptionsResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+// Delete deletes cluster options.
+func (r *clusterOptionsResource) Delete(
+	ctx context.Context,
+	req resource.DeleteRequest,
+	resp *resource.DeleteResponse,
+) {
 	var state clusterOptionsModel
 	diags := req.State.Get(ctx, &state)
 	resp.Diagnostics.Append(diags...)
 
-	if resp.Diagnostics.HasError() {
-		return
+	var toDelete []string
+
+	if !state.Keyboard.IsUnknown() && state.Keyboard.ValueString() != "" {
+		toDelete = append(toDelete, "keyboard")
+	}
+
+	if state.bandwidthData() != "" {
+		toDelete = append(toDelete, "bwlimit")
+	}
+
+	if state.crsData() != "" {
+		toDelete = append(toDelete, "crs")
+	}
+
+	if state.haData() != "" {
+		toDelete = append(toDelete, "ha")
+	}
+
+	if state.migrationData() != "" {
+		toDelete = append(toDelete, "migration")
+	}
+
+	if state.EmailFrom.ValueString() != "" {
+		toDelete = append(toDelete, "email_from")
+	}
+
+	if state.Language.ValueString() != "" {
+		toDelete = append(toDelete, "language")
+	}
+
+	if state.Console.ValueString() != "" {
+		toDelete = append(toDelete, "console")
+	}
+
+	if !state.HTTPProxy.IsUnknown() && state.HTTPProxy.ValueString() != "" {
+		toDelete = append(toDelete, "http_proxy")
+	}
```

Comment:
why sometimes you use `IsUnkown()` and sometimes not? For quite identical in behavior string attributes like `HTTPProxy` and `Console`.

Reply:
Ah, that's an oversight 😅, thanks for the review btw!

I don't check for `IsUnkown()` in case of composite params like `if state.migrationData() != ""`, as the check is inside, but for primitives the check is needed.
Also, at delete time all values will be known, so the check should be actually using `IsNull()`.